initial umd changes

j-5-s opened the pull request:
This seems to work with both a `<script>` tag for cdn as well as standard es imports. address #1

j-5-s committed:
initial umd changes

j-5-s commented on webpack.config.js:
```diff
@@ -7,7 +7,7 @@ const moduleOverrides = {
 
 const nullExternals = {
-    resolve: 'window.resolve',
+    // resolve: 'window.resolve',
```
im not sure what this was for but after changing the libraryTarget to umd i was getting a react error when trying to import directly via a create-react-app i was testing with.

mhsdesign replied:
these are for not including stuff we dont need, to lower the bundle size...

were you importing in your app this lib via `dist` or from `source`?

j-5-s replied:
I was linked via `yarn link` so `dist/main.js` through the package.json.

Steps to recreate:

1. Clone [this test repo](https://github.com/jamescharlesworth/tailwind-react-app-test). Its a simple create-react-app that [imports](https://github.com/jamescharlesworth/tailwind-react-app-test/blob/main/src/App.js#L2) `browser-jit-generate-tailwindcss`.
2. Checkout the [umd-build branch](https://github.com/jamescharlesworth/browser-jit-generate-tailwindcss/tree/umd-build) of your repo i forked.
3. Uncomment [the line](https://github.com/jamescharlesworth/browser-jit-generate-tailwindcss/blob/umd-build/webpack.config.js#L10) and run the build
4. `yarn link` (from `umd-branch` branch of `browser-jit-generate-tailwindcss`)
5. `yarn link browser-jit-generate-tailwindcss` (from my react test repo)
6. `npm start` from my react test repo

Expected result: it should work without errors
Actual result: error is thrown:
```
ERROR in ../browser-jit-generate-tailwindcss/dist/main.js 4:82-107
Module not found: Error: Can't resolve 'window.resolve' in '/Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/dist'
resolve 'window.resolve' in '/Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/dist'
  Parsed request is a module
  using description file: /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/package.json (relative path: ./dist)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/dist/node_modules doesn't exist or is not a directory
      looking for modules in /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/node_modules
        single file module
          using description file: /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/package.json (relative path: ./node_modules/window.resolve)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/node_modules/window.resolve doesn't exist
            .web.mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jamescharlesworth/projects/jvf/test/browser-jit-generate-tailwindcss/node_modules/window.resolve.web.mjs doesn't exist
...
```

mhsdesign replied:
nice thanks for the steps ;)

we should get rid of the line!

dynamic imports work too (in a bundle) ;) and will be split by your demo react app automatically in a chunk and loaded seperately on demand nice!

`<script type=module>` with import and dynamic imports in the browser dont work, because there is no `default` export available but one could use them like this:

```
await import('../dist/main.js')
console.log(window.browserJitGenerateTailwindcss)
```

j-5-s replied:
awesome, thanks! I also am gonna look at adding support for typescript this week. I think it was throwing a typescript error when I tried it out on a create-react-app with typescript.

mhsdesign committed:
TASK: Dont replace `resolve` with null

Module not found: Error: Can't resolve 'window.resolve' in '/this/dist'

when using this in a webpack app

mhsdesign committed:
TASK: Fix Style; Use LF and 4 indents, editor config
